 surfaceflinger: Validate setposition parameters

Validate setposition parameters width and height to
fix hang issues becuase of wrong values set by clients.

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index 7607b42..d1a9cb0 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -59,6 +59,8 @@
 
 #define DEBUG_RESIZE    0
 
+#define MAX_POSITION 32767
+
 namespace android {
 
 // ---------------------------------------------------------------------------
@@ -1653,6 +1655,10 @@ uint32_t Layer::setTransactionFlags(uint32_t flags) {
 bool Layer::setPosition(float x, float y, bool immediate) {
     if (mCurrentState.requested.transform.tx() == x && mCurrentState.requested.transform.ty() == y)
         return false;
+    if ((y > MAX_POSITION) || (x > MAX_POSITION)) {
+        ALOGE("%s:: failed %s  x = %f y = %f",__FUNCTION__,mName.string(),x, y);
+        return false;
+    }
     mCurrentState.sequence++;
 
     // We update the requested and active position simultaneously because
